### DISCUSSION TITLE
FastRelax specific chains/residues

### DISCUSSION THREAD

**Discussion by pjmbro** [RosettaCommons/rosetta#280] [RosettaScripts]
I am trying to perform FastRelax on only chain H in an antibody-antigen complex with chains B, H and L. Due to the large numbers of jumps (chain B has a large number of glycans) I opted to use MoveMapFactory to restrict any DOF in bb and chi, after which i allow only chi sampling in chain H. However, after opening the ouput I find that also in chain B and L, sidechain chi torsions have moved. Any clue what is going wrong here, or how i can restrict DOF in chi to only chain H (or any other selection from RESIDUE_SELECTORS)? Thanks a lot for any help!
```
ROSETTASCRIPTS>
    <SCOREFXNS>
        <ScoreFunction name="sfxn_relax" weights="beta_nov16" >
            <Reweight scoretype="arg_cation_pi" weight="3" />
            <Reweight scoretype="approximate_buried_unsat_penalty" weight="5" />
            <Set approximate_buried_unsat_penalty_burial_atomic_depth="3.5" />
            <Set approximate_buried_unsat_penalty_hbond_energy_threshold="-0.5" />
        </ScoreFunction>
    </SCOREFXNS>
    <RESIDUE_SELECTORS>
        <Chain name="chainH" chains="H"/>
    </RESIDUE_SELECTORS>
    <MOVE_MAP_FACTORIES>
        <MoveMapFactory name="movemap_H" bb="0" chi="0" jumps="0">
            <Chi enable="1" residue_selector="chainH"/>
        </MoveMapFactory>
    </MOVE_MAP_FACTORIES>
    <MOVERS>
            <FastRelax name="fast_relax" scorefxn="sfxn_relax" repeats="1" cartesian="false" ramp_down_constraints="false" min_type="dfpmin_armijo_nonmonotone" movemap_factory="movemap_H"/>
    </MOVERS>
    <PROTOCOLS>
        <Add mover="fast_relax"/>
    </PROTOCOLS>
</ROSETTASCRIPTS>
```

**Comment by roccomoretti** [comment]
FastRelax is a combination of minimization and sidechain repacking. The MoveMap controls how the DOFs change during the minimization phase, but don't (necessarily) control what happens during the repacking stage. Instead, it's the task_operations which control which sidechains are allowed to repack.

I'd recommend using a [NotResidueSelector](https://docs.rosettacommons.org/docs/latest/scripting_documentation/RosettaScripts/ResidueSelectors/ResidueSelectors#residueselectors_logical-residueselectors_notresidueselector) with your ChainSelector in combination with the [PreventRepacking](https://docs.rosettacommons.org/docs/latest/scripting_documentation/RosettaScripts/TaskOperations/taskoperations_pages/PreventRepackingOperation) task operation to turn off repacking on the non-H chains.

**Reply by pjmbro** [reply]
Thanks Rocco for the quick reply!
I now added TaskOperations that should prevent repacking in the non-H chains. However, I am still seeing different rotamers than the rotamers from the input pdb for these chains (B and L). Perhaps the MoveMapFactory is not functioning as it should, still allowing Chi minimization in those chains? I can clearly see it hasn't repacked non-H chains though by the looks of the clashing sidechains in chains B and L. 
```
<SCOREFXNS>
        <ScoreFunction name="sfxn" weights="beta_nov16">
            <Reweight scoretype="coordinate_constraint" weight="1.0" />
        </ScoreFunction>
        <ScoreFunction name="sfxn_relax" weights="beta_nov16">
            <Reweight scoretype="coordinate_constraint" weight="1.0" />
            <Reweight scoretype="arg_cation_pi" weight="3" />
            <Reweight scoretype="approximate_buried_unsat_penalty" weight="5" />
            <Set approximate_buried_unsat_penalty_burial_atomic_depth="3.5" />
            <Set approximate_buried_unsat_penalty_hbond_energy_threshold="-0.5" />
        </ScoreFunction>
    </SCOREFXNS>

    <RESIDUE_SELECTORS>
        <Chain name="chainH" chains="H" />
        <Chain name="chainL" chains="L" />
        <Chain name="chainB" chains="B" />
    </RESIDUE_SELECTORS>

    <TASKOPERATIONS>
        <LimitAromaChi2 name="limitchi2" chi2max="110" chi2min="70" include_trp="True" />
        <ExtraRotamersGeneric name="ex1_ex2" ex1="1" ex2="1" />
        <IncludeCurrent name="current" />
        <OperateOnResidueSubset name="no_repackB" selector="chainB">
            <PreventRepackingRLT />
        </OperateOnResidueSubset>
        <OperateOnResidueSubset name="no_repackL" selector="chainL">
            <PreventRepackingRLT />
        </OperateOnResidueSubset>
    </TASKOPERATIONS>

    <MOVE_MAP_FACTORIES>
        <MoveMapFactory name="movemap_H" bb="0" chi="0" jumps="0">
            <Chi enable="1" residue_selector="chainH" />
        </MoveMapFactory>
    </MOVE_MAP_FACTORIES>

    <MOVERS>
        <FastRelax name="fast_relax" scorefxn="sfxn_relax" repeats="3" cartesian="false" ramp_down_constraints="false" min_type="dfpmin_armijo_nonmonotone" task_operations="limitchi2,ex1_ex2,current,no_repackB,no_repackL" movemap_factory="movemap_H"/>
    </MOVERS>

    <PROTOCOLS>
        <Add mover="fast_relax" />
    </PROTOCOLS>    `
```
